server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceShardsAllocator.java

```diff
@@ -193,6 +198,15 @@ public void allocate(RoutingAllocation allocation, ActionListener<Void> listener
         reconcile(currentDesiredBalance, allocation);
     }
 
+    private void processNodeShutdowns(ClusterState clusterState) {
+        final var nodeShutdowns = clusterState.metadata().nodeShutdowns().getAllNodeIds();
+        if (nodeShutdowns.equals(processedNodeShutdowns) == false) {
```

Comment:
I do not think we need to reset when removing shutdown entry for the node that is already removed. This is also not going to harm anything.

Do we need to reset for the restart case?